pkg/tsdb/azuremonitor/loganalytics/azure-log-analytics-datasource.go

```diff
@@ -172,6 +172,9 @@ func buildLogAnalyticsQuery(query backend.DataQuery, dsInfo types.DatasourceInfo
 	basicLogsEnabled := false
 
 	resultFormat := ParseResultFormat(azureLogAnalyticsTarget.ResultFormat, dataquery.AzureQueryTypeAzureLogAnalytics)
+	if azureLogAnalyticsTarget.ResultFormat != nil {
```

Comment:
I think you can remove these lines as you're doing this check in `ParseResultFormat`